### PR TITLE
Add NodeList.item()

### PR DESCRIPTION
**Pull request by rickycodes**
https://developer.mozilla.org/en-US/docs/Web/API/NodeList/item

Not sure how useful this is...

Maybe not very?

### CODE REVIEW

**Comment by Pauan** [comment]
This needs to return `Option< Node >`

Also it should use 4 spaces for indentation.

**Comment by rickycodes** [comment]
@Pauan updated!

**Comment by Pauan** [comment]
@rickycodes Awesome, thanks!

**Comment by Diggsey** [comment]
Can we use the `Index` trait for this?

**Comment by Pauan** [comment]
@Diggsey `Index` doesn't return `Option`, instead it panics. So we would ideally want both.

**Comment by Diggsey** [comment]
Never mind - due to lack of HKT the `Index` trait was designed to always return a &reference, which is not possible here.

File: src/webapi/node_list.rs [hunk 1]
```diff
@@ -30,6 +30,16 @@ impl NodeList {
         js!( return @{self}.length; ).try_into().unwrap()
     }
 
+    /// Returns a node from a NodeList by index.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/NodeList/item)
+    // https://dom.spec.whatwg.org/#dom-nodelist-item
```
Comment (koute) [inline src/webapi/node_list.rs]: This should link to the IDL: `https://dom.spec.whatwg.org/#ref-for-dom-nodelist-item`

**Comment by rickycodes** [comment]
@koute: updated! (:

**Comment by koute** [comment]
LGTM; thanks!

**Pull request merged**